### PR TITLE
Feature/game speed

### PR DESCRIPTION
**Pull request by Rinkeeee96**

### CODE REVIEW

**Comment by Rinkeeee96** [comment]
image

**Review by wieltar — DISMISSED** [review]

File: Glitch/Game/Commands/GlobalCommands/DecreaseGameSpeedCommand.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include "Game/Commands/ICharacterCommand.h"
+#include "Game/Commands/GlobalCommand.h"
+#include "Events/Action/DecreaseGameSpeedEvent.h"
+
+class DecreaseGameSpeedCommand: public GlobalCommand
```
Comment (Rinkeeee96) [inline Glitch/Game/Commands/GlobalCommands/DecreaseGameSpeedCommand.h:6]: DOXYGENNNNN!!!

File: Glitch/Game/Commands/GlobalCommands/IncreaseGameSpeedCommand.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include "Game/Commands/ICharacterCommand.h"
+#include "Game/Commands/GlobalCommand.h"
+#include "Events/Action/IncreaseGameSpeedEvent.h"
+
+class IncreaseGameSpeedCommand : public GlobalCommand
```
Comment (Rinkeeee96) [inline Glitch/Game/Commands/GlobalCommands/IncreaseGameSpeedCommand.h:6]: DOXYGENNNN!!!

File: engineFoxtrot/Engine/Events/Action/DecreaseGameSpeedEvent.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include "Events/Event.h"
+
+class DecreaseGameSpeedEvent : public Event
```
Comment (Rinkeeee96) [inline engineFoxtrot/Engine/Events/Action/DecreaseGameSpeedEvent.h:4]: DXOYEGNNNENNENENENENENXDOXYYEGNNENNN

File: engineFoxtrot/Engine/Events/Action/IncreaseGameSpeedEvent.h [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include "Events/Event.h"
+
+class IncreaseGameSpeedEvent : public Event
```
Comment (Rinkeeee96) [inline engineFoxtrot/Engine/Events/Action/IncreaseGameSpeedEvent.h:4]: DXOYYYGGGGGGGGGEEEEEEEEEEEENNNNNNNNNNN

File: engineFoxtrot/Engine/Fps/FrameData.cpp [hunk 5]
```diff
@@ -21,40 +22,54 @@ void FrameData::startTimer() {
...
-	if (reset)
-		this->startTimer();
-	else
-		this->fps = this->calculateAverageFps();
-	reset = !reset;
-}
+	return (static_cast<const float>(lastFrameFps != 0 ? timeStep / lastFrameFps : 1)) * multiplier;
+}
+
+void FrameData::setEventDispatcher(EventDispatcher* _dispatcher)
```
Comment (Rinkeeee96) [inline engineFoxtrot/Engine/Fps/FrameData.cpp:28]: DOXYGENNNNNNNNNNNNNNNNNNNNNNNN

**Review by ttleeuw — DISMISSED** [review]

**Review by Maxvani — APPROVED** [review]

**Review by ttleeuw — APPROVED** [review]

**Pull request merged**